std/common.sy

```diff
@@ -17,6 +14,6 @@ split: pu str -> [str] : external
 
 as_float: pu int -> float : external
 as_int: pu float -> int : external
-as_char: pu str -> int : external
+as_char: fn str -> int : external
 as_chars: pu str -> [int] : external
```

Comment:
this feels a bit off

Reply:
Hmm, i changed it but a problem is that if the string is empty it actually returns nil rather than an int, so it does not necessarily return an int.

Reply:
That feels like a problem with the implementation not the purity. If it doesn't modify the string I see this as being pure. I'd rather see we change the return-type of the function to use the Maybe type from the standard library. ;)